### PR TITLE
Check Role in Private Route

### LINKED ISSUE #291: Modify PrivateRoute to check for user role
Add an enum to AuthTypes.ts in `frontend\src\types\AuthTypes.ts` for Role:
```
export enum Role {
  Admin = "Admin",
  Volunteer = "Volunteer",
  Employee = "Employee",
}
``` 

Change the role field of `AuthenticatedUser` in `frontend\src\types\AuthTypes.ts` to be of type Role. Idk if that works, but if it doesn't then change it to `"Admin" | "Volunteer" | "Employee"` . 

Modify `PrivateRoute` in `frontend\src\components\auth\PrivateRoute.tsx` to check the user's role. Add a new prop called `authorizedRoles?` of type Role[]. Provide a default value of []. If the authorizedRoles is empty or if the user's role matches a role in authorizedRoles, allow access. Else, redirect. 

Go to `frontend\src\App.tsx` and add the new prop to each instance of `PrivateRoute`. Routes prefaced with VOLUNTEER should require the `volunteer` role and routes prefaced with ADMIN require the `admin` role. Otherwise, do not require any roles.

### PR DESCRIPTION
**Pull request by hujoseph99**
## Ticket link

Closes #291 

## Implementation description
* Add role checks for the different routes on FE

## Steps to test
Looks at the routes at frontend\src\constants\Routes.ts
Sign in as volunteer.
1. check that you cannot access routes that start wth `/admin`
2. Check that you can access all other routes

Sign in as admin
1. Check that you cannot access routes that start with `/volunteer`
2. Check that you can access all other routes

## What should reviewers focus on?
* Checking to see if the routes behave as expected

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/types/AuthTypes.ts [hunk 1]
```diff
@@ -1,9 +1,15 @@
+export enum Role {
+  Admin = "ADMIN",
```
Comment (hujoseph99) [inline frontend/src/types/AuthTypes.ts:2]: Keep casing consistent with ticket description. For example `Admin = "Admin"`

File: frontend/src/components/auth/PrivateRoute.tsx [hunk 2]
```diff
@@ -3,21 +3,28 @@ import { Route, Redirect } from "react-router-dom";
...
   path: string;
   exact: boolean;
+  authorizedRoles?: Role[];
 };
 
 const PrivateRoute: React.FC<PrivateRouteProps> = ({
   component,
   exact,
   path,
+  authorizedRoles,
```
Comment (hujoseph99) [inline frontend/src/components/auth/PrivateRoute.tsx:19]: Should make this prop default to `[]`

**Review by hujoseph99 — COMMENTED** [review]
Looks good to me! Just a few nits

File: frontend/src/types/AuthTypes.ts (hunk 1 above)
Comment (LenaNguyen) [reply frontend/src/types/AuthTypes.ts:2]: the role returned by the API is in all caps actually. Seems easier to make the enum all caps as well then having to change it to uppercase when doing a string compare

**Comment by hujoseph99** [comment]
LGTM!

**Review by LenaNguyen — APPROVED** [review]

**Pull request merged**